### ISSUE TITLE
Textpattern 4.8.5 release flight plan

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1594]
We are targeting Textpattern 4.8.5 for release in Q1/Q2 2021 to address issues relating to PHP 8.0 as outlined in https://github.com/textpattern/textpattern/issues/1592, plus other snagging.

* [4.8.5 branch](https://github.com/textpattern/textpattern/tree/4.8.5)

Please update this post and subsequent replies with tasks etc as needed, assigning yourself or others where appropriate.

**Repo Issues**

* [`textpattern/textpattern`](https://github.com/textpattern/textpattern/milestone/18)
* [`textpattern/textpattern.github.io`](https://github.com/textpattern/textpattern.github.io/milestone/2)

## Personnel

* Release pilot: @Bloke 
* Release co-pilot: @petecooper 

## Preflight

- [x] preflight
  - [x] latest translations (@philwareham)
  - [x] latest pophelp (@philwareham)
  - [x] draft .com blog post (release pilot, or assigned person)
  - [x] write change history (everyone)
  - [x] tag public/admin themes with release tag (@philwareham)
  - [x] tag Textpacks and Pophelps with release tag (@philwareham)

## Take Off

- [x] take off
  - [x] package up for GitHub and .com (release pilot)
  - [x] update `latest.zip` and `latest.tar.gz` in server config (@petecooper)
  - [x] update language pack links in server config (@petecooper)
  - [x] update demo site (@petecooper)
  - [x] publish .com blog post (release pilot, or assigned person)
  - [x] forum announcement (release pilot, or assigned person)
  - [x] forum feedback (release pilot, or assigned person)

## Inflight service

- [x] inflight service
  - [x] close 4.8.5 milestone (release pilot, or assigned person)
  - [x] update `TXPversion` on `rpc.textpattern.com` (release pilot, or assigned person)
  - [x] update https://en.wikipedia.org/wiki/Template:Textpattern_version (@petecooper or @philwareham)
  - [x] socials (@petecooper or @philwareham)
  - [x] repin https://github.com/textpattern/textpattern/issues/279 (anyone)

**Comment by Bloke** [comment]
I'll make a branch now. Makes sense to mop these niggles up.

**Comment by petecooper** [comment]
How are we looking for a December patch release, sirs?

The milestone queue has a bunch of @Bloke or @bloatware tasks at the moment, mindful of that and potential feature creep that might compound the issue further.

**Comment by philwareham** [comment]
Fine by me, although my tasks for this release are minimal.

**Comment by bloatware** [comment]
I won't have much free time until the end of the year, sorry, so my vote is blank.

**Comment by Bloke** [comment]
Doable.

**Comment by petecooper** [comment]
Couple of stinkers removed from the milestone, only thing outstanding now is https://github.com/textpattern/textpattern/issues/1585

I'll make a start on the .com blog post draft now.

**Comment by petecooper** [comment]
Tentatively reassigning @bloatware given recent commits.

**Comment by bloatware** [comment]
I have installed php 8 and immediately run into a bug. Our testing policy is to tighten.

**Comment by petecooper** [comment]
@bloatware - what did you find?

**Comment by bloatware** [comment]
Silenced erroneous calls to `checkdate()` function used in `year_month_day_title` mode that are no more silenced in php 8.

**Comment by jools-r** [comment]
One more to-do for "Inflight service":

- [ ] Update TXPversion on rpc.textpattern.com

**Comment by philwareham** [comment]
I've tagged the themes and languages with 4.8.5 now, so my part is done. Best wishes for a nice Christmas all!

**Comment by petecooper** [comment]
I think we're close. Release post drafted, will need another look with fresh eyes: ID 435 if you're curious, please edit mercilessly.

**Comment by Bloke** [comment]
Great post, thanks. I just added the note about the diag panel (issue #1608). Are there any more HISTORY entries to go in? Think we've got them all.

**Comment by petecooper** [comment]
@bloatware @Bloke @philwareham

*Go* (thumbs up) or *No Go* (thumbs down) on 4.8.5 release, please.

Do we need another week or two to iron anything out?

**Comment by bloatware** [comment]
A week would be fine for me.

**Comment by Bloke** [comment]
Yeah, let's give it a few days for the dust to settle and get some hardcore testing done. Minor potential issue with e9a77f45 regarding single-digit months/days (introduced in 4.8.1) - see commit comment - but this change has given us an opportunity to fix my oversight from back then.

**Comment by petecooper** [comment]
@bloatware @Bloke @philwareham

Go (thumbs up) or No Go (thumbs down) on 4.8.5 release, please.

How long do we need to assign for testing?

**Comment by philwareham** [comment]
No additions from me.

**Comment by bloatware** [comment]
If nothing is broken with my recent break tweaks, I'd stop tweaking. If you happen to utilize global `trim/replace` attributes, please check whether everything works as expected.

**Comment by petecooper** [comment]
OK, we have a Go (thank you, all) for release.

What are the key points for user testing before we ship? I have a bit of time later to put some words together and we can rouse some testers in the usual places. If we have focus areas for them to look at, I suspect that will yield better results.

I've noted `trim` / `replace` attributes as mentioned in https://github.com/textpattern/textpattern/issues/1594#issuecomment-762404966 - anything further to add to the laundry list?

**Comment by Bloke** [comment]
All good from my perspective. Thank you everyone for all the work you've put into this release.

I would encourage prople to give the front end tags a good grilling to check the parser still behaves as expected. In particular not only tags that work but also to bend them and try misspelled attributes (to check they'd still rejected and throw an appropriate error in debug/testing mode) and tags-in-atts and txp:yield and short codes and all that goodness. Plus checking if the trim/replace work as expected and the global `break`. 

The other main thing is also the PHP capabilities. Want to make sure that there's now a reduced attack surface for PHP exploits by non-trusted users, taking into account that Content Editors can now use PHP in articles. In theory, anyone should be able to use shortcodes in articles, regardless of their permissions, even if those forms use txp:php tags. But they should not be allowed to pass PHP into the forms unless they have article level permissions to do so.

**Comment by petecooper** [comment]
[RFC]

We should make a .com blog post with a request for more eyes, and outline some of the things that have changed.

Given we have no super-urgent issues to fix in userland that a week or so would exacerbate, I think we'd benefit from a code freeze, a switch into testing mode for a week to ten days, *then* barring any snags that need fixing, we release.

So, broadly:

* new / optimised code freeze now (or as near as possible)
* blog post goes live weekend of Jan 23 & 24 2021 (this weekend)
* at least a full week of testing from interested parties, issues opened where appropriate, fixes permitted
* assuming no major work required, prepare for release Feb 3 or 4 2021

This release may end up being the final 4.8 release, so it would be good to shake out as much as we can move on to 4.9…and a week or so would be a good move, I think.

Thoughts? Pros? Cons?

**Comment by Bloke** [comment]
Perfect. I'm not adding anything else to 4.8.5 so a freeze works for me. That means playing in 4.9 this week instead, yay.

**Comment by bloatware** [comment]
Frozen, up to cleanup.

**Comment by petecooper** [comment]
I've started a blog post for a 4.8.5 testers call (ID 436). There are some things to add in, particularly the testing focus areas (PHP in articles, short codes and so forth), I'll have another pass later this afternoon after my meetings.

Please edit mercilessly and without prejudice. I've taken a fairly informal tone to ensure we send the right message, but extra eyeballs are appreciated as always.

**Comment by bloatware** [comment]
Thanks, Pete. I would be glad to help with technicalities, but do not seem to have an account. Let me in :-)

**Comment by Bloke** [comment]
I can make you an account. What access level do you need? ME okay?

**Comment by bloatware** [comment]
Thanks, just read is okay.

**Comment by Bloke** [comment]
Hmm, you've already got an account. I sent a password reset but didn't get the green notification that it got sent. Maybe we don't have mail() installed or something. Weird. Let me know by PM if you don't get it and we'll figure it out off-air.

**Comment by petecooper** [comment]
Mail server is up and running:

![Screenshot 2021-01-25 at 13 42 52](https://user-images.githubusercontent.com/226578/105713621-46b1b200-5f13-11eb-93d2-d0ceecc15bca.png)

**Comment by bloatware** [comment]
Got mail, thanks. I have a working account on plugins, but found no traces of the root site. Everything is ok now, thanks.

**Comment by Bloke** [comment]
Weird. I used multi-edit 'Reset password' from the Users panel and... crickets. And this in the Console:

```
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“style-src”).
```

**Comment by petecooper** [comment]
I'll take https://github.com/textpattern/textpattern/issues/1594#issuecomment-766831683 - can you throw an issue into https://github.com/textpattern/server-config/issues please and I'll deal. Thanks!

**Comment by Bloke** [comment]
Magic, thank you. [Issue raised](https://github.com/textpattern/server-config/issues/57).

**Comment by petecooper** [comment]
Apologies for radio silence this month, some trying situations offline. I'm hoping to be able to finish the article outlined in https://github.com/textpattern/textpattern/issues/1594#issuecomment-766806592 _today_, and if it passes muster with enough eyeballs here, plus any late-breaking areas of interest from your point of view, we can publish and publicise (publishise?) for a greater reach. Gut feeling says a few weeks of testing would be well-spent, then we get into launch mode for mid-February.

**Comment by petecooper** [comment]
OK, I've completed the second draft of the call for testers article (436), I'll take a final look with new eyes at about 6pm. If anyone wants to jump in and tweak, please do. Could do with a appropriate `h3.` near the beginning to balance it a bit, but I think the tone is OK overall.

Barring any massive hiccups, I'll publishise™ tonight at 8pm-ish and see what the next two weeks brings our way!

**Comment by Bloke** [comment]
Thank you! I've added a few things to test. Can be tidied up if you wish as they're a bit wordy.

**Comment by petecooper** [comment]
Perfect – thanks v much, @Bloke. I've done final tweaks, will set to publishise™ around 8pm so you have a few hours to tweak.

**Comment by petecooper** [comment]
Is there further ongoing work underway with 4.8.5 that warrants more time, or are we broadly happy to proceed to release?

**Comment by philwareham** [comment]
I’m happy to release. No work needed from me.

**Comment by bloatware** [comment]
No work planned for the next two weeks on my side. If the tests are over, go for release.

**Comment by Bloke** [comment]
I still haven't got a PHP 8 environment. Plucking up the courage to rebuild my server droplet and will build PHP 8 there. But subject to people being happy that we've caught all the PHP 8 issues from the previous release, there's no fallout from the tightened `<txp:php>` implementation, and the tag attributes are still doing their thing correctly, I'm happy to go for 4.8.5 in its current guise.

**Comment by petecooper** [comment]
OK, I'll take some more PHP 8 testing today and see if I can break anything. Assuming no headline faults, we can shoot for a release during the week if that's viable

@Bloke - how's your availability this week? Do you want to take release engineer role for this one?

**Comment by Bloke** [comment]
Sure, perfect, I can take this one no worries.

**Comment by petecooper** [comment]
Final call for any `HISTORY.txt` updates / changes, please. I can go through commits this weekend but youse folks are far more familiar with the work you've done for this release, so please be patient if I ask you anything!

**Comment by petecooper** [comment]
Right, I can't find anything in commits that's not covered by current entries (at least the stuff I understand), and I was not able to break it with PHP 8.0.2…so, soft thumbs up here. I'll mark `HISTORY.txt` item as done.

**Comment by petecooper** [comment]
@Bloke – I have some capacity today and tomorrow if you're happy with testing outcomes, can assist with the release buttons etc.

**Comment by philwareham** [comment]
I've put the latest translations from Crowdin into the 4.8.5 branch now. Release whenever you are ready to. Cheers!

**Comment by petecooper** [comment]
How are we placed for a weekend release @Bloke? Mindful it's half term etc.

**Comment by Bloke** [comment]
I'm out of coverage tomorrow all day, but will be back late afternoon/evening for further testing. Could do Sunday sometime.

**Comment by philwareham** [comment]
Translations are fully up-to-date so it's all done on my side. I will tag the textpack and pophelp repos with `4.8.5` after you've released.

**Comment by philwareham** [comment]
Bump? :)

**Comment by Bloke** [comment]
Sorry, weekend was a bust. Testing today.

**Comment by petecooper** [comment]
I'm trying to understand where the MAMP issues as outlined in https://github.com/textpattern/textpattern/issues/1642 and https://github.com/textpattern/textpattern/issues/1633 fit into our 4.8.5 release intentions. I don't know the extent of the work that's gone into mitigating it so far, beyond some defensive coding in the setup. How big a deal is this? Is 4.8.5 realistic for a 'fix', if indeed it is a Textpattern issue?

**Comment by Bloke** [comment]
I'm happy that adding the 'en' entry does "fix" the issue for MAMP. It's our only supported language that didn't explicitly have a locale defined in our code. Perhaps we just got away with it by chance till now and we should have had that all along, rather than relying on the system to provide it? Who knows. But I'm happy it works and it feels safer.

**Comment by petecooper** [comment]
Thanks @Bloke - makes sense now.

**Comment by philwareham** [comment]
Yep, what @bloke said. Fixed well enough for me.

**Comment by philwareham** [comment]
I just have someone translating the Textpacks into Fulah language which will take a couple of days more. After that I think we should release 4.8.5 and move on ASAP.

**Comment by petecooper** [comment]
>I just have someone translating the Textpacks into Fulah language 

…which put Textpattern ahead of Wordpress in terms of Fulah translation! Amazing!

4.8.5 issue queue is clear as of now.

![Tg8pZ5R](https://user-images.githubusercontent.com/226578/109963156-fda80700-7ce3-11eb-84e7-45693d91d712.gif)

**Comment by philwareham** [comment]
The Fulah language Textpacks are done. The pophelp is going to take a few weeks so I'd say go ahead and release 4.8.5 now if you are ready to.

**Comment by petecooper** [comment]
I'll update OP to mention the RSS bug mentioned in https://forum.textpattern.com/viewtopic.php?id=51405 since there doesn't appear to be an issue to track as yet.

**Comment by petecooper** [comment]
@Bloke @bloatware – given recent commits, are there any particular areas we need to test before we can let 4.8.5 loose?

**Comment by bloatware** [comment]
@petecooper yes, please, syndication feeds and `<txp:variable />` with all attributes combinations on my side.

**Comment by petecooper** [comment]
I haven't yet found any quirks or bugs in my testing, though it shouldn't be considered comprehensive.

@Bloke @bloatware - anything else in the pipeline from either of you, or can we proceed to release mode? Thank you.

**Comment by bloatware** [comment]
@petecooper everything looks nailed, no pending requests, what else? Let's turn the 4.8 page.

**Comment by petecooper** [comment]
@Bloke - how are you placed for a release today or tomorrow? I can run it if you're busy, no trouble.

**Comment by Bloke** [comment]
Just heavy tag testing left to go, really. The interface seems to be holding up okay, and the installation issues have been nailed as far as I can tell, though we need to run through some update scenarios from suitable older versions to be sure.

Today or tomorrow looks doable.

**Comment by petecooper** [comment]
Are there any missing `HISTORY.txt` talking points for parser optimisations that we've baked in? I can see commits that have the sprinkling of etcMagic© but I'm not clever enough to know what's changed.

**Comment by bloatware** [comment]
Well, can not totally exclude that some minor magic/security fixes/Easter eggs are undocumented, but this would not be the first time. Nothing deserving a mention in annals, imo.

But we should incite people to set `name` attribute of `<txp:header />` before they update. It is no more `Content-Type` by default, and failing this would break custom headers.

**Comment by bloatware** [comment]
@petecooper perfect, thank you.

**Comment by petecooper** [comment]
Forum migrated without issue, seems to be faster, stable, email is flowing freely (especially to @phiw13), so green light from the infrastructure department.

**Comment by jools-r** [comment]
Big thanks once again! Saw the tweet too. Impressive!.

Just one thing: I see `Maintenance: Sorry, we are currently performing maintenance, please come back later.` 

No worries if you're still tweaking things, but otherwise maybe “maintenance mode" is still on?

**Comment by petecooper** [comment]
> Just one thing: I see `Maintenance: Sorry, we are currently performing maintenance, please come back later.` 

Likely stale DNS, Jools - try clearing cache and local DNS, failing that try again Monday AM.

**Comment by jools-r** [comment]
> > Just one thing: I see `Maintenance: Sorry, we are currently performing maintenance, please come back later.`
> 
> Likely stale DNS, Jools - try clearing cache and local DNS, failing that try again Monday AM.

Thanks. It must be that but I clearing browser caches and flushing Mac dns cache in the terminal makes no different. Pinging forum.textpattern.com gives me `46.101. …` while pinging tornado.textpattern.net gives me `68.183. …` so it's obviously not made the jump yet. Weirdly I do see it on my smartphone although it is on the same Wifi network/provider and not on the mobile network. I don't get that.

**Comment by philwareham** [comment]
OK, can we release or is there issues preventing? Ta.

**Comment by bloatware** [comment]
The only open issue tagged 4.8.5 is this one. No commit planned on my side.

**Comment by petecooper** [comment]
I'll sanity check the release blog post to make sure we're up to date on the changes made.

**Comment by Bloke** [comment]
@bloatware Could you please give me some examples of using the `root` attribute in `<txp:page_url />`. Nothing I've tried so far seems to output anything if I use it either valueless or with a value. Scratching my head to work out what it does. Thanks!

**Comment by bloatware** [comment]
@Bloke ah, sorry. The idea is to customize permlinks, to make them relative (with `root`) or set/inherit some URL parameters (with `context`):
```
<txp:page_url root context="lang" lang="default">
    <!-- will create relative permlinks and set or inherit lang parameter -->
    <txp:permlink />
    <!-- /path/to/root/my-article?lang=inherited|default -->
</txp:page_url>
```

**Comment by Bloke** [comment]
Ahhh, sweet. Nice example, thank you. I was just bamboozled as it didn't seem to affect any of the output from the tag itself and I was expecting it to return relative links (or absolute links to a nominated domain if a `root` value was supplied) for the _existing_ things that could be returned via the `type` attribute. I see now that was a foolish expectation :)

I'll add some tag constructs like that to the unit tests later.

**Comment by bloatware** [comment]
> relative links (or absolute links to a nominated domain if a root value was supplied) for the existing things that could be returned via the type attribute

What kind of things, exactly?

Probably, `root="/"` is more appropriate than valueless `root` as shortcut for `/path/to/root`. I don't see under which circumstances one would need to construct `/my-article` links, that would then be resolved as `http(s)://www.example.com/my-article`. Though URL rewriting, maybe?

**Comment by Bloke** [comment]
That's what I was thinking, for URL rewriting purposes and what I expected the `root="https://some-domain/"` attribute to do when the tag was used as a single. Hence my confusion. Never thought to try it as a container - neat!

One other thing: when you say 'inherit' lang parameter... That example always outputs `?lang=default` because that's what's set in the tag, even if I supply `?lang=fr` in the page URL. That's expected, I guess because the inline attribute overrides it, yes?

So, to grab the current value and use it, we could do this?

``` html
<txp:variable name="thislang" trim>
    <txp:page_url type="lang" default="en" />
</txp:variable>
<txp:page_url root context="lang" lang='<txp:variable name="thislang" />'>
   <txp:permlink />
</txp:page_url>
```

**Comment by Bloke** [comment]
I'm not saying we should change this now, btw. Wanna get 4.8.5 out today if we can. It works spiffingly well as it is and we can look into tweaking it in 4.9.

**Comment by bloatware** [comment]
Hmm, something looks wrong, let me see.

**Comment by bloatware** [comment]
Well, maybe it's ok, just not intuitive enough (for me). If we inverted the priority (inherited, then default), it would be tricky to output a chunk of `en` text on `?lang=fr` page. You can spare a variable though:
```
<txp:page_url root context="lang" lang='<txp:page_url type="lang" default="en" />'>
   <txp:permlink />
</txp:page_url>
```
And if you are on `page_url` docs page, please check [this example](https://docs.textpattern.com/tags/page_url#example-6-link-to-the-latest-article-in-a-section), it looks like `most-recent` should be replaced with `latest` there.

**Comment by bloatware** [comment]
Also, this will inherit `lang` only when it is set:
```
<txp:page_url context="lang">
   <txp:permlink />
</txp:page_url>
```

**Comment by Bloke** [comment]
Super. It's fine as-is. Makes sense to me.

And I've fixed that example: good catch, thanks.

**Comment by Bloke** [comment]
Oooh, one other thing - and this might just be me - but do you consider the output of `<txp:page_url>` to be "wrong" (currently, and perhaps since time immemorial) when Txp is installed as a subdir?

If you install Txp at http://localhost:888/4.8.5 and issue a bare `<txp:page_url>` tag in a non-textiled article, you see this:

/4.8.5/articles/my-test-article

It's correct insofar as that is the actual URL but the '/4.8.5' forms part of the `siteurl` in prefs. Should (could?) it be omitted here if we set some attribute to tell it to omit it? Or is this correct behaviour do you think?

Maybe in future versions we might consider:

`<txp:page_url />` => /4.8.5/articles/my-test-article (for backwards compat)
`<txp:page_url root />` => /articles/my-test-article
`<txp:page_url root="http://example.org" />` => http://example.org/articles/my-test-article
`<txp:permlink />` => https://example.com/4.8.5/articles/my-test-article
`<txp:permlink root />` => /articles/my-test-article

Or something like that?

**Comment by bloatware** [comment]
Dunno whether it was "wrong", but a bare `<txp:page_url />` (without `context`) just outputs the corresponding `$pretext` field value, ignoring `root`. We could tweak it, but it seems that `<txp:page_url context root="/" />` already works as you describe: `/articles/my-test-article` URL. And `<txp:page_url context root="" />` outputs `articles/my-test-article`. Both are not directly usable as `href`: the first one will be resolved as `http://localhost:8888/articles/my-test-article`, and the second one as `http://localhost:8888/articles/my-test-article/articles/my-test-article` by the browser.

I guess we can postpone tweaking to 4.9, this does not look as release blocker?

**Comment by Bloke** [comment]
Yeah, you're right. It's fine as is. Just me not thinking it through. Full steam ahead.

**Comment by bloatware** [comment]
If you have not yet calculated checksums and all, I can make `root` trigger URL mode, no problem.

**Comment by Bloke** [comment]
Calculating checksums can be done at any time - one-line command. I've not done the "this is 4.8.5" commit yet.

But you're right that maybe it doesn't make sense to do root URLs from the `<txp:page_url>` tag, since the URL it would spit out isn't usable if you're in a subdir. Well, maybe it is but only if you want to construct them by hand:

```<txp:site_url /><txp:page_url root />```

Dunno. If you think it has value, by all means add it. But I'm not sure if adding it would cause confusion, since it would only apply in some situations, right? (i.e. for some values of `type`).

**Comment by bloatware** [comment]
Yep, `root` and `type` do not cohabit well. Ok, let us leave it for 4.9.

**Comment by Bloke** [comment]
@bloatware Is this expected with the new article custom `pageby`?

1. On the front (default) page of a new install (random article content populated via cbe_rand_content), edit the `<txp:article limit="5" />` tag and swap it for `<txp:article_custom pageby limit="5" />`
2. All looks good.
3. Scroll down. Click 'Older'.
4. ?pg=2 appears in the URL but the list of articles is the same as on page 1.

Additional intriguing side effect:

5. Remove `pageby`.
6. Refresh page.
7. Note error message: 

```
<txp:newer rel="prev">
                                <txp:text item="newer" />
                            </txp:newer> ->  Textpattern Warning: secondpass < 2 while parsing form None on page default

Tag error: <txp:older rel="next">
                                <txp:text item="older" />
                            </txp:older> ->  Textpattern Warning: secondpass < 2 while parsing form None on page default
```
Doesn't matter what value of secondpass you set in advanced pref, the error message reports that it's less than one greater than that value.

Presumably I'm using it wrong or need to do something else to get the front page to paginate in this manner?

**Comment by bloatware** [comment]
It's neither expected nor unexpected - just a decision to make. Should `<txp:article_custom pageby />` output obey `pg` parameter, à la `<txp:article />`? I don't remember why `pageby` was added, probably a forum question that has been solved this way. Looking at the code, it is possible that `<txp:article_custom pageby offset />` does what you need, but for some reason the pagination has not been made automatic.

Re `newer/older` the behaviour looks normal. They need pagination data and postpone themselves to the next pass if it is not available. Since no tag sets pagination, at the last pass an error message is issued.

**Comment by Bloke** [comment]
Fine by me. If it's by design then great. We'll just need to ensure that people know they can't use `pg` and will have to wire up pagination by hand, or use the valueless `offset` trick which does - after rudimentary testing - appear to do the trick. Nice one!

**Comment by bloatware** [comment]
Since we still have this `missing article tag` message, I suppose that `article_custom` can not yet fully replace `article`, though they are getting closer and closer. Is there a use-case where paginated `<txp:article_custom />` is really handy?

**Comment by Bloke** [comment]
Honestly don't know. I'm sure in the past I've wished it could be paginated but right now I can't think of when that was.

**Comment by Bloke** [comment]
Okay, we're done. 4.8.5 is live on GitHub. Cigar hovering in front of lips, awaiting a light.

Give or take, I'm up to [step 21 of the release process](https://github.com/textpattern/textpattern-toolbelt). Things left to do by anyone whose eyes aren't shutting of their own accord every five minutes:

* Create forum posts, Twitter announcement etc.
* Alter .com article URL to forum posts and publish it.
* Figure out why the 4.8.5 branch can't be deleted from the repo via the command line. I get `error: dst refspec 4.8.5 matches more than one. error: failed to push some refs to 'https://github.com/textpattern/textpattern'. If someone can please do it by hand from GitHub at some point, that'd be most appreciated.
* Update RPC/.com (latest version) and server config stuff. Side note: toolbelt article link to server config in Step 22 needs updating as it 404s after server move.
* Update the docs, heh, especially [tag docs](https://forum.textpattern.com/viewtopic.php?id=51386).

As always, I am indebted to all your due diligence, professionalism and downright greatness at putting up with my dithering and whimsy nature firing off ideas that aren't always well thought-out. Thank you so much for shaping this release and keeping all these plates spinning aboard the good ship Textpattern.

**Comment by petecooper** [comment]
The day shift will take dibs from here!

**Comment by petecooper** [comment]
OK, all done barring the docs. Closing. Please re-open if needed.
